app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,25 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def expired_token; end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user 
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
This construction is not 100% ok, because when user enters invalid password you are not handling it. Instead you should think about using `update` instead of set and save.

```ruby
if user && user.update(password: params[:password]) 
  user.generate_token
  # ...
```